sample-docker-app/index.ts

```diff
@@ -1,41 +1,57 @@
-import { Pond, Semantics, OnStateChange, Subscription, FishTypeImpl } from '@actyx/pond'
+import { Pond, Fish, FishId, Tag } from '@actyx/pond'
 
 // Each fish keeps some local state it remembers from all the events it has seen
 type State = { time: string, name: string, msg: string, } | undefined
+const nameTag = Tag<string>('name')
 
-const ForgetfulChatFish: FishTypeImpl<State, string, string, State> = FishTypeImpl.of({
+const mkForgetfulChatFish = (name: string): Fish<State, string> => ({
     // The kind of fish is identified by the meaning of its event stream, the semantics
-    semantics: Semantics.of('ForgetfulChatFish'),
+    fishId: FishId.of('ForgetfulChatFish', name, 0),
 
     // When the fish first wakes up, it computes its initial state and event subscriptions
-    initialState: (_name, _sourceId) => ({
-        state: undefined, // start without information about previous event
-        subscriptions: [Subscription.of(ForgetfulChatFish)] // subscribe across all names
-    }),
+    initialState: undefined, // start without information about previous event
 
     // Upon each new event, keep some details of that event in the state
-    onEvent: (_state, event) => ({
-        time: new Date(event.timestamp / 1000).toISOString(),
-        name: event.source.name,
-        msg: event.payload
-    }),
+    onEvent: (_state, event, metadata) =>
+        ({
+            time: metadata.timestampAsDate().toISOString(),
+            name: metadata.tags.find(x => x.startsWith(
+                `${nameTag.rawTag}:`
+            ))?.split(':')[1] || ":-(",
```

Comment:
I think we want to have a convenience function in the pond to extract those tags.

Reply:
The guidance is that you should never need to dissect tags, because every piece of data should also be part of the event proper.

It would be nice if we could get around this potential duplication, by doing it the way proper DB does: Typed columns and then indices based on the columns. But we’re a long way from that…

Reply:
Ah, thanks for the reminder, yes I agree. So @wngr please disregard my remark on the Tag extraction :-)

Reply:
Alright, added the sender into the event.